File: src/main/java/net/reflact/engine/spells/dynamic/DamageEffect.java
package net.reflact.engine.spells.dynamic;

import com.google.gson.JsonObject;
import net.minestom.server.entity.Entity;
import net.minestom.server.entity.LivingEntity;
import net.reflact.common.attribute.RpgAttributes;
import net.reflact.engine.ReflactEngine;
import net.reflact.engine.data.ReflactPlayer;

public class DamageEffect implements SpellEffect {
    private double baseDamage;
    private double scaling;

    @Override
    public void execute(CastContext context) {
        Entity target = context.getTarget();
        if (target instanceof LivingEntity living) {
            double damage = baseDamage;
            
            // Apply Intelligence Scaling
            ReflactPlayer casterData = ReflactEngine.getPlayerManager().getPlayer(context.getCaster().getUuid());
            if (casterData != null) {
                double intel = casterData.getAttributes().getValue(RpgAttributes.INTELLIGENCE);
                damage += (intel * scaling);
            }
            
            living.damage(net.minestom.server.entity.damage.DamageType.MAGIC, (float) damage);
        }
    }

    @Override
    public void load(JsonObject config) {
        this.baseDamage = config.has("damage") ? config.get("damage").getAsDouble() : 5.0;
        this.scaling = config.has("scaling") ? config.get("scaling").getAsDouble() : 0.0;
    }
}

File: src/main/java/net/reflact/engine/spells/dynamic/DamageEffect.kt
package net.reflact.engine.spells.dynamic

import com.google.gson.JsonObject
import net.minestom.server.entity.LivingEntity
import net.reflact.common.attribute.RpgAttributes
import net.reflact.engine.ReflactEngine

class DamageEffect : SpellEffect {
    private var baseDamage: Double = 0.0
    private var scaling: Double = 0.0

    override fun execute(context: CastContext) {
        val target = context.target
        if (target is LivingEntity) {
            var damage = baseDamage

            // Apply Intelligence Scaling
            val casterData = ReflactEngine.getPlayerManager().getPlayer(context.caster.uuid)
            if (casterData != null) {
                val intel = casterData.attributes.getValue(RpgAttributes.INTELLIGENCE)
                damage += (intel * scaling)
            }

            target.damage(net.minestom.server.entity.damage.DamageType.MAGIC, damage.toFloat())
        }
    }

    override fun load(config: JsonObject) {
        this.baseDamage = if (config.has("damage")) config.get("damage").asDouble else 5.0
        this.scaling = if (config.has("scaling")) config.get("scaling").asDouble else 0.0
    }
}
